nipype/interfaces/ants/preprocess.py

```diff
@@ -0,0 +1,355 @@
+        transform_str = "-t {}[{}]"
+        if (isdefined(self.inputs.transformation_model)
+                and isdefined(self.inputs.gradient_step_length)):
+            return transform_str.format(self.inputs.transformation_model,
+                                        self.inputs.gradient_step_length)
+        raise ValueError("Unable to format transformation_model argument")
+
+    def _format_output(self):
+        if (isdefined(self.inputs.output_transform_prefix)
+                and isdefined(self.inputs.output_warped_image)
+                and isdefined(self.inputs.output_average_image)):
+            return "-o [{},{},{}]".format(
+                self.inputs.output_transform_prefix,
+                self.inputs.output_warped_image,
+                self.inputs.output_average_image
+            )
+        elif isdefined(self.inputs.output_average_image):
+            return "-o {}".format(self.inputs.output_average_image)
+        else:
+            raise ValueError("Unable to format output due to lack of inputs.")
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        if isdefined(self.inputs.output_average_image):
+            outputs['average_image'] = (
+                os.path.abspath(self.inputs.output_average_image)
+            )
+        if isdefined(self.inputs.output_warped_image):
+            outputs['warped_image'] = (
+                os.path.abspath(self.inputs.output_warped_image)
+            )
+        if isdefined(self.inputs.output_transform_prefix):
+            fname = '{}MOCOparams.csv'.format(
+                self.inputs.output_transform_prefix
+            )
+            outputs['composite_transform'] = os.path.abspath(fname)
+        if (isdefined(self.inputs.write_displacement) and
+                isdefined(self.inputs.output_transform_prefix) and
+                self.inputs.write_displacement is True):
+            fname = '{}Warp.nii.gz'.format(self.inputs.output_transform_prefix)
+            outputs['displacement_field'] = os.path.abspath(fname)
+        return outputs
+
+class Matrix2FSLParamsInputSpec(BaseInterfaceInputSpec):
+    matrix = File(
+        exists=True,
+        desc='Motion crrection matrices to be converted into FSL style motion parameters',
+        mandatory=True
+    )
+
+class Matrix2FSLParamsOutputSpec(TraitedSpec):
+    parameters = File(exists=True, desc="parameters to be output")
+
+
+class Matrix2FSLParams(BaseInterface):
+    '''
+    Take antsMotionCorr motion output as input, convert to FSL style
+    parameter files. Currently does not output origin of rotation.
+    '''
+    input_spec = Matrix2FSLParamsInputSpec
+    output_spec = Matrix2FSLParamsOutputSpec
+
+    def _run_interface(self, runtime):
+        in_fp = open(self.inputs.matrix)
+        in_data = csv.reader(in_fp)
+        pars = []
+
+        # Ants motion correction output has a single line header that we ignore
+        next(in_data)
+
+        for x in in_data:
```

Comment:
is this doing the same [`nibabel.euleragnles.mat2euler`](http://nipy.org/nibabel/reference/nibabel.eulerangles.html#nibabel.eulerangles.mat2euler) is doing?